### ISSUE TITLE
org.hibernate.MappingException: Unknown collection role when filtering by `.size` for certain associations

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#518]
Affected properties:

 - `analysis.experimentAnalyzed.characteristics.size`
 - `analysis.experimentAnalyzed.otherRelevantPublications.size`
 - `experimentalFactors.size`

Those will be excluded from filterable properties.

**Comment by arteymix** [comment]
Fixed in 572d97fac3d69e7daff63476d7367102f2f702a7

**Comment by arteymix** [comment]
Nevermind, this is not fixed in https://github.com/PavlidisLab/Gemma/commit/572d97fac3d69e7daff63476d7367102f2f702a7.

**Comment by arteymix** [comment]
This is caused by the alias substitution algorithm resulting in `.size` being applied to a non-collection alias.

Example:


Currently:

```sql
select * from entity e
join e.characteristics as c
where c.size
```

Instead it should be:

```sql
select * from entity e
join e.characteristics as c
where characteristics.size
```

**Comment by ppavlidis** [comment]
Closing during 2.0-alpha bug triage as **won't fix / fixed**. arteymix's last comment confirmed the fix landed in f9be37d3c0 ("Omit the .size suffix when looking up a suitable alias") — that commit is present on `phase2-acl-migrate`.

**Comment by ppavlidis** [comment]
Gemma 2 disposition: Won't fix
